code/datums/components/simple_atom_storage.dm

```diff
@@ -0,0 +1,68 @@
+/// Simple component for handling storage of items into a simple atom.
+/datum/component/simple_atom_storage
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/list/atom/movable/stored_items = list()
+	var/list/allowed_types
+	var/maximum_slots
+	var/sound_file
+
+/datum/component/simple_atom_storage/Initialize(list/allowed_types, maximum_slots, sound_file)
+	. = ..()
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	src.allowed_types = allowed_types
+	src.maximum_slots = maximum_slots
+	src.sound_file = sound_file
```

Comment:
Hey, for my own information could you explain why you'd use `src.` here wouldn't it be implied?

Reply:
Because the proc argument variable has the same name
So src.var refers to the datum one while the second one refers to the proc arg one
It's a pattern that seems used a lot in constructors here, i think because it avoids having to make up slightly different variable names which can be confusing

Reply:
Brilliant thank you 👍

Reply:
> Brilliant thank you 👍

fira is correct, and it's explicitly requested in our style guide too (https://github.com/cmss13-devs/cmss13/blob/master/.github/guides/STYLES.md#when-passing-vars-through-new-or-initializes-arguments-use-srcvar)